src/appleseed/foundation/curve/binarycurvespecs.txt

```diff
@@ -0,0 +1,87 @@
+            Specifications of the BinaryCurve file format
+                    Revision 1 - May 28th, 2018
+
+
+
+INTRODUCTION
+
+  The purpose of the BinaryCurve file format is to store 3D curve geometry in a compact
+and efficient-to-read form.
+
+  The format is fully LITTLE-ENDIAN, regardless of the machine used to author
+files.
+
+
+
+GENERAL STRUCTURE
+
+  .----------------------------------.
+  |             Signature            |    11 bytes (string without 0 at the end)
+  +----------------------------------+
+  |              Version             |    2 bytes (16-bit unsigned integer)
+  +----------------------------------+
+  |               Data               |
+  `----------------------------------'
+
+  The signature field must contain the 11-character long string "BINARYCURVE".
+If it contains any other value, the file is not a valid BinaryCurve file.
+
+  The format of the Data block depends on the value of the Version field.
+
+
+
+DATA BLOCK FORMAT VERSION 1
+
+
+  .----------------------------------.
+  |             Basis                |    1 byte (8-bit unsigned char enum - linear(1)/ bezier(2)/ b-spline(3)/ catmull-rom(4))
+  +----------------------------------+
+  |         Number of curves         |    4 bytes (32-bit unsigned integer)
+  +----------------------------------+
+  |    Number of vertices curve #1   |    4 bytes (32-bit unsigned integer)
+  +----------------------------------+
+  | Coordinates of vertex #1 [X,Y,Z] |    12 bytes (3 single precision floats)
+  +----------------------------------+
+  | Coordinates of vertex #2 [X,Y,Z] |    12 bytes (3 single precision floats)
+  +----------------------------------+
+  | Coordinates of vertex #3 [X,Y,Z] |    12 bytes (3 single precision floats)
+  +----------------------------------+
+  |              ...                 |
+  +----------------------------------+
+  |        Width at vertex #1        |    4 bytes (single precision float)
+  +----------------------------------+
+  |        Width of vertex #2        |    4 bytes (single precision float)
+  +----------------------------------+
+  |              ...                 |
+  +----------------------------------+
+  |    Transparency at vertex #1     |    4 bytes (single precision float)
```

Comment:
Sometimes (i.e. in the API) you talk about opacity, here you talk about transparency. Which is it? Make consistent.